app/scripts/app-init.js

```diff
@@ -111,14 +107,20 @@ function importAllScripts() {
   }
 }
 
+// Ref: https://stackoverflow.com/questions/66406672/chrome-extension-mv3-modularize-service-worker-js-file
 // eslint-disable-next-line no-undef
 self.addEventListener('install', importAllScripts);
 
 /*
- * Message event listener below loads script if they are no longer available.
+ * A keepalive message receiver to avoid Service Worker getting shut down due to inactivity.
+ * UI sends the message periodically.
+ *
  * chrome below needs to be replaced by cross-browser object,
  * but there is issue in importing webextension-polyfill into service worker.
  * chrome does seems to work in at-least all chromium based browsers
  */
-// eslint-disable-next-line no-undef
-chrome.runtime.onMessage.addListener(importAllScripts);
+// eslint-disable-next-line
+chrome.runtime.onMessage.addListener((message) => {
+  importAllScripts();
+  return false;
```

Comment:
I originally suggested to put something like  `return message.name === WORKER_KEEP_ALIVE_MESSAGE;` here because it'd help people find out where the message is coming from. I see a change where you remove the `const WORKER_KEEP_ALIVE_MESSAGE` line and replaced it with `return false;`. To be clear, there's no reason this function needs to return boolean, it was just means to get `WORKER_KEEP_ALIVE_MESSAGE` here. If you don't want that, there's no need to have the return statement.

Reply:
`return false` works well

I removed `return message.name === WORKER_KEEP_ALIVE_MESSAGE` as it is throwing error if this function returns true.

Reply:
It was not about returning anything, but pointing out the fact that message.name should be `WORKER_KEEP_ALIVE_MESSAGE` to the person wondering where the message is coming from. returning anything was not the point :) 

If it didn't work as an explanation for you, it's not gonna work on others :sweat_smile:

Reply:
I've added a suggestion to the comment above. Feel free to remove the return statement or keep it - whichever you find less surprising.

Reply:
I think what @naugtur is trying to say is that it's not straightforward to know what this event listener is listening to. If someone were to read this file, they would need to put a debugger here and hope it lands (which we happen to know it currently does every second), or try to interpret it from the comment above which doesn't provide a silver bullet answer either. I think it's also better to enable the code to speak for itself rather than adding comments.


What if we did something similar to the `updateBackgroundConnection` logic below?
something like
```
if (message.name === 'WORKER_KEEP_ALIVE_MESSAGE') { return; }

console.warn('ServiceWorker: unknown message handled by onMessage event: ', message.name);
```